snapcraft/lifecycle.py

```diff
@@ -101,6 +101,12 @@ def _run_step(self, step, part, part_names, dirty, recursed):
                 '{}'.format(part.name, ' '.join(prereqs)))
             self.run('stage', prereqs, recursed=True)
 
+        # Prepare to run this step, if such a method is implemented
+        try:
```

Comment:
@kyrofa what do you think is cleaner? I sort of like it here as it reduces code duplication (and if the other stuff moves here it will be a much smaller `PluginHandler` class.

Just say nay if you think it is not a good idea :smiley: